crates/sequencer/src/state.rs

```diff
@@ -277,4 +325,56 @@ mod tests {
         // When
         state.get_compiled_class_hash(*ONE_CLASS_HASH).unwrap();
     }
+
+    #[test]
+    pub fn dump_and_load_state() {
+        let mut state = State::default();
+
+        // setting up entry for state.classes
+        let class_hash = *ONE_CLASS_HASH;
+        let contract_class = include_str!("./test_data/cairo_0/compiled_classes/counter.json");
+        let contract_class: ContractClassV0 =  serde_json::from_str(contract_class).expect("failed to deserialize ContractClass from ./crates/sequencer/test_data/cairo_1/compiled_classes/account.json");
+        let contract_class = ContractClass::V0(contract_class);
+
+        let compiled_class_hash = *ONE_COMPILED_CLASS_HASH;
+        let contract_address = *TEST_CONTRACT;
+        let contract_storage_key: ContractStorageKey = (contract_address, *TEST_STORAGE_KEY);
+        let storage_value = *ONE_FELT;
+        let nonce = *TEST_NONCE;
+
+        state.classes.insert(class_hash, contract_class);
+        state
+            .compiled_class_hashes
+            .insert(class_hash, compiled_class_hash);
+        state.contracts.insert(contract_address, class_hash);
+        state.storage.insert(contract_storage_key, storage_value);
+        state.nonces.insert(contract_address, nonce);
+
+        let dump_file_path = PathBuf::from("./src/test_data/katana_dump.json");
+
+        state
+            .dump_state_to_file(&dump_file_path)
+            .unwrap_or_else(|error| {
+                panic!(
+                    "failed to save state to path {:?},\n error {:?}",
+                    dump_file_path, error
+                )
+            });
+
+        let loaded_state = State::load_state_from_file(&dump_file_path).unwrap_or_else(|error| {
+            panic!(
+                "failed loading state from path {:?},\n error {}",
+                dump_file_path, error
+            )
+        });
```

Comment:
I think it would be fine to `expect` here

Reply:
right right, I mean more in the sense do we need to format the error with the file path + this extra text? for me just `.expect("failed loading state from path") should give enough info.

Reply:
resolved via [this](https://github.com/kkrt-labs/ef-tests/pull/555/commits/5484da1997d8dbbe67c69ce05d0a7579b6106d94) commit.